test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java

```diff
@@ -329,14 +331,20 @@ private void populateCallbacks(ClassLoader classLoader) throws ClassNotFoundExce
         // make sure that we start over everytime we populate the callbacks
         // otherwise previous runs of QuarkusTest (with different TestProfile values can leak into the new run)
         quarkusTestMethodContextClass = null;
-        beforeAllCallbacks = new ArrayList<>();
+        afterConstructCallbacks = new ArrayList<>();
+        legacyAfterConstructCallbacks = new ArrayList<>();
         beforeEachCallbacks = new ArrayList<>();
         afterEachCallbacks = new ArrayList<>();
 
-        ServiceLoader<?> quarkusTestBeforeAllLoader = ServiceLoader
+        ServiceLoader<?> quarkusTestAfterConstructLoader = ServiceLoader
+                .load(Class.forName(QuarkusTestAfterConstructCallback.class.getName(), false, classLoader), classLoader);
+        for (Object quarkusTestAfterConstructCallback : quarkusTestAfterConstructLoader) {
+            afterConstructCallbacks.add(quarkusTestAfterConstructCallback);
+        }
+        ServiceLoader<?> quarkusTestLegacyAfterConstructLoader = ServiceLoader
                 .load(Class.forName(QuarkusTestBeforeAllCallback.class.getName(), false, classLoader), classLoader);
-        for (Object quarkusTestBeforeAllCallback : quarkusTestBeforeAllLoader) {
-            beforeAllCallbacks.add(quarkusTestBeforeAllCallback);
+        for (Object quarkusTestLegacyAfterConstructCallback : quarkusTestLegacyAfterConstructLoader) {
+            legacyAfterConstructCallbacks.add(quarkusTestLegacyAfterConstructCallback);
```

Comment:
I suppose we cannot (yet) log something here?

Reply:
I've never been able to log anything from the extension exception with System.out...